File: mixin-base-test/src/test/java/net/yakclient/asm/classloader/CustomClassLoader.java
package net.yakclient.asm.classloader;

import java.io.IOException;

public class CustomClassLoader extends ClassLoader {
    public Class<?> defineClass(String name, byte[] b) {
        return super.defineClass(name, b, 0, b.length);
    }

    @Override
    protected Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException {
        return super.loadClass(name, resolve);
    }

    public Class<?> loadSomeClass(byte[] b, String name) throws IOException {
        return this.defineClass(name, b);
    }
}

File: mixin-base-test/src/test/kotlin/net/yakclient/asm/classloader/CustomClassLoader.kt
package net.yakclient.asm.classloader

import java.io.IOException

class CustomClassLoader : ClassLoader() {
    fun defineClass(name: String?, b: ByteArray): Class<*> {
        return super.defineClass(name, b, 0, b.size)
    }

    @Throws(ClassNotFoundException::class)
    override fun loadClass(name: String, resolve: Boolean): Class<*> {
        return super.loadClass(name, resolve)
    }

    @Throws(IOException::class)
    fun loadSomeClass(b: ByteArray, name: String?): Class<*> {
        return this.defineClass(name, b)
    }
}